js/src/hooks/useEffectRemoveNotice.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { dispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import useNotices from '.~/hooks/useNotices';
+import { STORE_KEY } from '.~/data/constants';
+
+/**
+ * Search for a notice with specific label and remove it if the component is unmounted.
+ *
+ * @param {string} label the notice label
+ * @param {string} [storeKey] the store
+ *
+ * @return {import('@wordpress/notices').Notice|null} The notice to be removed otherwise null if it is not found
+ */
+const useEffectRemoveNotice = ( label, storeKey = STORE_KEY ) => {
```

Comment:
❓ 
Only the store keys of `core/notices` and `core/notices2` have the `removeNotice` action. Using the `STORE_KEY` constant as the default value will cause errors if the component sides omit `storeKey`.